### ISSUE TITLE
Support Java Modules

### ISSUE THREAD

**Issue by Quinn-With-Two-Ns** [temporalio/sdk-java#1940]
**Is your feature request related to a problem? Please describe.**
Support Java Modules

**Describe the solution you'd like**
Currently the Java SDK cannot be used as a named Java module since some of modules implement the same packages.

```
error: the unnamed module reads package io.temporal.internal.common from both temporal.sdk and temporal.serviceclient
error: the unnamed module reads package io.temporal.internal from both temporal.sdk and temporal.serviceclient
```

Java modules do not allow this, it is called a split package.

**Describe alternatives you've considered**
Don't support Jave modules

**Comment by Riva2006** [comment]
It is important fix to all Java developers ,  Currently we using shadow to repack the packages , It is important to get basic support of module-info

**Comment by dbiere** [comment]
This issue was opened nearly a year ago and is still a problem as of Temporal Java SDK 1.24.1. Is this on the roadmap to resolve? Thank you.

**Comment by Quinn-With-Two-Ns** [comment]
Yes we are aware of this issue, but I have no timeline on when this will be resolved.
